frontend/tests/blocks.spec.ts

```diff
@@ -51,160 +51,181 @@ for (const realmType of realms) {
-                description: "Cat software running on dog hardware.",
-            };
-            await test.step("Series can be changed", async () => {
-                const input = page.getByRole("combobox");
-                await input.pressSequentially(newSeries.title);
-                await page.getByRole("option", { name: newSeries.title }).click();
-                await saveButton.click();
-
-                await expect(page.getByRole("heading", { name: newSeries.title })).toBeVisible();
-            });
-
-            await test.step("Video order can be changed", async () => {
-                await editButton.click();
-                await page.getByLabel("Oldest first").setChecked(true);
-                await expect(page.getByLabel("Oldest first")).toBeChecked();
-                await saveButton.click();
-
-                await expect(page.getByRole("button", { name: "Choose video order" }))
-                    .toHaveText("Oldest first");
-            });
+                await test.step("Title block can be changed", async () => {
+                    const newTitle = "Nice test page";
+                    const editButton = page
+                        .getByRole("heading", { name: "Nice title" })
+                        .locator("..")
+                        .locator("..")
+                        .locator("..")
+                        .getByLabel("Edit block");
 
-            await test.step("Layout options can be changed", async () => {
-                await test.step("Title and videos (initial)", async () => {
-                    await expect(page.getByRole("heading", { name: newSeries.title }))
-                        .toBeVisible();
-                });
-
-                await test.step("Videos only", async () => {
                     await editButton.click();
-                    await page.getByLabel("Show title").setChecked(false);
-                    await page.getByLabel("Show description").setChecked(false);
+                    await page.getByPlaceholder("Title").fill(newTitle);
                     await saveButton.click();
 
-                    await expect(page.getByRole("heading", { name: newSeries.title }))
-                        .not.toBeVisible();
-                    await expect(page.getByText(newSeries.description))
-                        .not.toBeVisible();
+                    await expect(page.getByText(newTitle)).toBeVisible();
                 });
 
-                await test.step("Description and videos", async () => {
+                await test.step("Text block be changed", async () => {
+                    const newText = "The lazy dog jumps over the quick brown fox.";
+                    const editButton = page
+                        .getByText("Nice text.")
+                        // This is getting ridiculous.
+                        .locator("..")
+                        .locator("..")
+                        .locator("..")
+                        .locator("..")
```

Comment:
yeah kind of :/ this might mean that Playwright wants to discourage doing that? Mhpf meh, this is hard. I mean they are right: the specific DOM structure can change at any time, meaning we would have to revise the tests.

Maybe a better way is to use sth like `page.locator("div").filter({ has: "Nice text." })` (which will likely return many results, all the parent divs) and then locate ... in there? Huff. But it needs to be the smallest div containing "Edit block". Graaa, I think I have to think about this again. Or maybe you can google around and find a clever way that gets rid of all these disadvantages?